src/utils.ts

```diff
@@ -1,8 +1,25 @@
 import { lstatSync, accessSync, constants, readFileSync } from "fs";
 import { createHash } from "crypto";
+import { tmpdir } from "os";
 import { join } from "pathe";
 import type { PackageJson } from "pkg-types";
 
+// Workaround for pnpm setting an incorrect `TMPDIR`.
+// Set `JITI_TMPDIR_KEEP` to a truthy value to disable this workaround.
+// https://github.com/pnpm/pnpm/issues/6140
+// https://github.com/unjs/jiti/issues/120
+export function getCacheDir() {
+  if (process.env.JITI_TMPDIR_KEEP) {
+    return join(tmpdir(), "node-jiti");
+  }
+
+  const currentTmpDir = process.env.TMPDIR;
+  delete process.env.TMPDIR;
```

Comment:
If I'm current this situation only happens when `TMPDIR === cwd()` right? (otherwise we can support normal use of `TMPDIR`